tsdb/raw.go

```diff
@@ -93,6 +93,7 @@ func (e *RawExecutor) execute(out chan *models.Row) {
 	// Keep looping until all mappers drained.
 	var err error
 	for {
+		timeout := time.After(30 * time.Second)
```

Comment:
Yeah, it was a hack like this that I suggested during the sync-up but
@pauldix was concerned about the idea.

On Monday, November 23, 2015, Paul Dix <notifications@github.com> wrote:

> In tsdb/raw.go
> <https://github.com/influxdb/influxdb/pull/4889#discussion_r45681522>:
>
> > @@ -93,6 +93,7 @@ func (e *RawExecutor) execute(out chan *models.Row) {
> >  	// Keep looping until all mappers drained.
> >  	var err error
> >  	for {
> > +		timeout := time.After(30 * time.Second)
>
> we should not have a timeout by default. Otherwise people that are doing
> really large queries are going to report this as a bug.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/influxdb/influxdb/pull/4889/files#r45681522>.
>